src/compile/cache/mod.rs

```diff
@@ -28,9 +27,25 @@ pub enum CacheError {
 	#[error(transparent)]
 	StripPrefixError(#[from] std::path::StripPrefixError),
 }
+// CacheDirNotSupported is a top level struct and not an enum variant because
+// it's converted elsewhere to other errors using [#from] and we want to be
+// as specific as possible
+#[derive(Error, Debug)]
+#[error("cache directory does not exist on this platform")]
+pub struct CacheDirNotSupported;
```

Comment:
Ok to make it a a seprarated struct.
But it still exists as a variant in `CacheError`. You should probably the lines 23-24

Reply:
I made `CacheError::CacheDirNotSupportedError(#[from] CacheDirNotSupported)` and used `cache_dir()` utility instead of `cache::cache_dir()`